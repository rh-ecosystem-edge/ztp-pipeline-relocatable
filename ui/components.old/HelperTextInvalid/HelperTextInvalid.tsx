import React from 'react';

import { ExclamationCircleIcon } from '@patternfly/react-icons';
import { global_danger_color_100 as dangerColor } from '@patternfly/react-tokens';

export const HelperTextInvalid: React.FC<{ id: string; validation?: string }> = ({
  id,
  validation,
}) =>
  validation ? (
    <div className="helper-text-invalid">
      <ExclamationCircleIcon color={dangerColor.value} />
      <span data-testid={id}>{validation}</span>
    </div>
  ) : null;
