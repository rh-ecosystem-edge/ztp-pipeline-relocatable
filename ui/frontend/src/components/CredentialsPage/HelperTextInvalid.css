.helper-text-invalid span {
  margin-left: var(--pf-global--spacer--sm);
}
