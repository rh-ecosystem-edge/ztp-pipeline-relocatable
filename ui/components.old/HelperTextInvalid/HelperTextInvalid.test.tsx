import React from 'react';
import { render } from '@testing-library/react';
import { HelperTextInvalid } from './HelperTextInvalid';

describe('HelperTextInvalid', () => {
  it('can render empty', () => {
    const { container } = render(<HelperTextInvalid id="my-id" />);
    expect(container).toMatchSnapshot();
  });

  it('can render with text', () => {
    const { container } = render(<HelperTextInvalid id="my-id" validation="My validation text" />);
    expect(container).toMatchSnapshot();
  });
});
