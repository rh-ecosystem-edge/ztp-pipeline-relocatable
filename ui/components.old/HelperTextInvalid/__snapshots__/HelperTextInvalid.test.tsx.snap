// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelperTextInvalid can render empty 1`] = `<div />`;

exports[`HelperTextInvalid can render with text 1`] = `
<div>
  <div
    class="helper-text-invalid"
  >
    <svg
      aria-hidden="true"
      fill="#c9190b"
      height="1em"
      role="img"
      style="vertical-align: -0.125em;"
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
      />
    </svg>
    <span
      data-testid="my-id"
    >
      My validation text
    </span>
  </div>
</div>
`;
